amqp/amqp.php

```diff
@@ -42,153 +42,178 @@
  */
-define('AMQP_NOWAIT', 8192);
+const AMQP_NOWAIT = 8192;
 
 /**
  * If set during a call to {@link AMQPQueue::nack()}, the message will be placed back to the queue.
  */
-define('AMQP_REQUEUE', 16384);
+const AMQP_REQUEUE = 16384;
 
 /**
  * A direct exchange type.
  */
-define('AMQP_EX_TYPE_DIRECT', 'direct');
+const AMQP_EX_TYPE_DIRECT = 'direct';
 
 /**
  * A fanout exchange type.
  */
-define('AMQP_EX_TYPE_FANOUT', 'fanout');
+const AMQP_EX_TYPE_FANOUT = 'fanout';
 
 /**
  * A topic exchange type.
  */
-define('AMQP_EX_TYPE_TOPIC', 'topic');
+const AMQP_EX_TYPE_TOPIC = 'topic';
 
 /**
  * A header exchange type.
  */
-define('AMQP_EX_TYPE_HEADERS', 'headers');
+const AMQP_EX_TYPE_HEADERS = 'headers';
+
+const AMQP_OS_SOCKET_TIMEOUT_ERRNO = 536870947;
+
+const PHP_AMQP_MAX_CHANNELS = 256;
+
+const AMQP_SASL_METHOD_PLAIN = 0;
+
+const AMQP_SASL_METHOD_EXTERNAL = 1;
 
 /**
- * The error number of OS socket timeout.
+ * Default delivery mode, keeps the message in memory when the message is placed in a queue.
  */
-define('AMQP_OS_SOCKET_TIMEOUT_ERRNO', 536870947);
+const AMQP_DELIVERY_MODE_TRANSIENT = 1;
 
 /**
- * The maximum number of channels that can be open on a connection.
+ * Writes the message to the disk when the message is placed in a durable queue.
  */
-define('PHP_AMQP_MAX_CHANNELS', 256);
+const AMQP_DELIVERY_MODE_PERSISTENT = 2;
 
 /**
- * SASL PLAIN authentication. This is enabled by default in the RabbitMQ server and clients, and is the default for most other clients.
+ * Extension version string
  */
-define('AMQP_SASL_METHOD_PLAIN', 0);
+const AMQP_EXTENSION_VERSION = '1.1.12alpha3';
 
 /**
- * Authentication happens using an out-of-band mechanism such as x509 certificate peer verification, client IP address range, or similar. Such mechanisms are usually provided by RabbitMQ plugins.
+ * Extension major version
  */
-define('AMQP_SASL_METHOD_EXTERNAL', 1);
+const AMQP_EXTENSION_VERSION_MAJOR = 0;
+
+/**
+ * Extension minor version
+ */
+const AMQP_EXTENSION_VERSION_MINOR = 1;
+
+/**
+ * Extension patch version
+ */
+const AMQP_EXTENSION_VERSION_PATCH = 12;
+
+/**
+ * Extension extra version suffix
+ */
+const AMQP_EXTENSION_VERSION_EXTRA = 'alpha3';
+
+/**
+ * Extension version ID
+ */
+const AMQP_EXTENSION_VERSION_ID = '10112';
 
 /**
  * stub class representing AMQPBasicProperties from pecl-amqp
  */
 class AMQPBasicProperties
 {
     /**
-     * @param string $content_type
-     * @param string $content_encoding
-     * @param array  $headers
-     * @param int    $delivery_mode
-     * @param int    $priority
-     * @param string $correlation_id
-     * @param string $reply_to
-     * @param string $expiration
-     * @param string $message_id
-     * @param int    $timestamp
-     * @param string $type
-     * @param string $user_id
-     * @param string $app_id
-     * @param string $cluster_id
+     * @param ?string $contentType
+     * @param ?string $contentEncoding
+     * @param ?string $correlationId
```

Comment:
PhpDoc misses info for $headers, $deliveryMode and $priority. But actually this PhpDoc doesn't bring any useful information since all types declaration and names of parameters are already available in signature so probably it makes sense to remove it completely. Could you please either update PHPDoc or remove it completely?

Reply:
Yes, you're right.